lib/train/options.rb

```diff
@@ -59,9 +59,6 @@ def merge_options(base, opts)
           default = hm[:default]
           if default.is_a? Proc
             res[field] = default.call(res)
-          elsif hm.key?(:coerce)
-            field_value = hm[:coerce].call(res)
-            res[field] = field_value.nil? ? default : field_value
```

Comment:
With this change, the user can no longer override the port and user on the command line. I understand not wanting to provide that functionality for all options, but removing the capability when it is already been provided is a breaking change. We should try to preserve this ability.